src/cascadia/TerminalApp/ActionArgs.h

```diff
@@ -36,30 +37,134 @@ namespace winrt::TerminalApp::implementation
     {
         CopyTextArgs() = default;
         GETSET_PROPERTY(bool, TrimWhitespace, false);
+
+        static constexpr std::string_view TrimWhitespaceKey{ "trimWhitespace" };
+
+    public:
+        static winrt::TerminalApp::IActionArgs FromJson(const Json::Value& json)
+        {
+            // LOAD BEARING: Not using make_self here _will_ break you in the future!
+            auto args = winrt::make_self<CopyTextArgs>();
+            if (auto trimWhitespace{ json[JsonKey(TrimWhitespaceKey)] })
+            {
+                args->_TrimWhitespace = trimWhitespace.asBool();
+            }
+            return *args;
+        }
     };
 
-    struct NewTabWithProfileArgs : public NewTabWithProfileArgsT<NewTabWithProfileArgs>
+    struct NewTabArgs : public NewTabArgsT<NewTabArgs>
     {
-        NewTabWithProfileArgs() = default;
-        GETSET_PROPERTY(int32_t, ProfileIndex, 0);
+        NewTabArgs() = default;
+        GETSET_PROPERTY(Windows::Foundation::IReference<int32_t>, ProfileIndex, nullptr);
+
+        static constexpr std::string_view ProfileIndexKey{ "index" };
+
+    public:
+        static winrt::TerminalApp::IActionArgs FromJson(const Json::Value& json)
+        {
+            // LOAD BEARING: Not using make_self here _will_ break you in the future!
+            auto args = winrt::make_self<NewTabArgs>();
+            if (auto profileIndex{ json[JsonKey(ProfileIndexKey)] })
+            {
+                args->_ProfileIndex = profileIndex.asInt();
+            }
+            return *args;
+        }
     };
 
     struct SwitchToTabArgs : public SwitchToTabArgsT<SwitchToTabArgs>
     {
         SwitchToTabArgs() = default;
         GETSET_PROPERTY(int32_t, TabIndex, 0);
+
+        static constexpr std::string_view TabIndexKey{ "index" };
+
+    public:
+        static winrt::TerminalApp::IActionArgs FromJson(const Json::Value& json)
+        {
+            // LOAD BEARING: Not using make_self here _will_ break you in the future!
+            auto args = winrt::make_self<SwitchToTabArgs>();
+            if (auto tabIndex{ json[JsonKey(TabIndexKey)] })
+            {
+                args->_TabIndex = tabIndex.asInt();
+            }
+            return *args;
+        }
+    };
+
+    // Possible Direction values
+    static constexpr std::string_view LeftString{ "left" };

```

Comment:
Shouldn't constant string patterns like this for parsing go wherever the rest of the parsing is for things like dimensions?